lib/server-manager.js

```diff
@@ -174,6 +175,13 @@ export class ServerManager {
     await Promise.all(this._activeServers.map(s => this.stopServer(s)));
   }
 
+  async restartAllServers(): Promise<void> {
+    this.stopListening();
+    await this.stopAllServers();
+    this._editorToServer = new Map();
```

Comment:
We should probably check to make sure we're not leaking anything - e.g. around the document text sync listeners and the diagnostics.

Reply:
All listeners are registered with this._disposable and stopListening disposes of it, startListening creates a new one for new listeners.  I think we're covered on that front!

Reply:
I double-checked this and found that we're shutting down the "exclusive" adapters correctly in ServerManager, but I just converted the linting and signature help adapters to be added to the server's disposables list so that we can be a little more sure about when they are disposed just by looking at the code.  Let me know if that's undesirable and I can revert that commit!